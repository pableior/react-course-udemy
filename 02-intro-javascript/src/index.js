
const firstName = 'Pablo';
const lastName = 'Gonzalez';
let diceValue = 5;

diceValue = 6;
console.log(firstName, lastName, diceValue);

if (true) {
    const firstName = 'Peter';
    let diceValue = 1;
    console.log(firstName, diceValue);
}

console.log(firstName, diceValue);

