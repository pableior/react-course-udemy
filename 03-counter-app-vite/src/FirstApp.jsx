import PropTypes from 'prop-types';
// yarn add prop-types

const sayHi = (name) => {
    return `Hola ${name}, buen dia!`;
}

const newMsg = {
    a: 'hola',
    b: 'Pablo',
}

export const FirstApp = ({title, subTitle, name}) => {

    return (
        <>
            {/* <h1> {sayHi('Pablo')} </h1> */}
            {/* <h1>{JSON.stringify(newMsg)}</h1> */}
            <h1>{title}</h1>
            <p> {subTitle} </p>
            <p> {name} </p>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Pablo Gonzalez',
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo', //va a dar error porque el tipo es del tipo number
}

