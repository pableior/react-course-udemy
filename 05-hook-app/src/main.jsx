import React from 'react'
import ReactDOM from 'react-dom/client'
import { BrowserRouter } from "react-router-dom";
import './index.css'
import { MainApp } from './09-useContext/MainApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { FocusScree } from './04-useRef/FocusScree'
// import { Layout } from './05-useLayoutEffect/Layout'
// import { Memorize } from './06-memos/Memorize'
// import { MemoHook } from './06-memos/MemoHook'
// import { CallbackHooks } from './06-memos/CallbackHooks'
// import { Padre } from './07-tarea/Padre'

//import './08-useReducer/intro-reducer';
//import { TodoApp } from './08-useReducer/TodoApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
  {/*<React.StrictMode>*/}
    <MainApp />
  {/*</React.StrictMode>,*/}
  </BrowserRouter>
)
