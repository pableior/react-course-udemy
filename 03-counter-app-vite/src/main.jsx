import React from 'react';

import ReactDOM from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
import './styles.css';
import { CounterApp } from './CounterApp';
ReactDOM.createRoot(document.getElementById('root')).render (
    <React.StrictMode>
        <FirstApp value = {15} />
    </React.StrictMode>
);