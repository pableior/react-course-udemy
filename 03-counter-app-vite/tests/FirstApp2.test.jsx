import { render, screen } from '@testing-library/react';
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en first app ', () => { 
    

    test('debe de hacer match con el snapshot', () => {
        
        const title = 'Hola, soy goku';
        const { container } = render( <FirstApp title={title}/>)

        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el msg hola soy goku', () => {
        
        const title = 'Hola, soy goku';
        render( <FirstApp title={title}/>)
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe de mostrar ekl titulo en un h1', () => {    
        const title = 'Hola, soy goku';
        render( <FirstApp title={title}/>)
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('debe de mostrar ekl subtitulo enviado por props', () => {    
        const title = 'Hola, soy goku';
        const subtitle = 'soy un subtitulo';

        render( <FirstApp title={title} subTitle={subtitle}/>);
        console.log(screen.getAllByText(subtitle));
        expect(screen.getAllByText(subtitle).length).toBe(1);
    });


    
 });