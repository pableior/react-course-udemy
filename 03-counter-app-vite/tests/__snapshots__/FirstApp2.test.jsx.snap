// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en first app  debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy goku
  </h1>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    Pablo Gonzalez
     
  </p>
</div>
`;
